server/ServerApp.cpp

```diff
@@ -3602,7 +3602,7 @@ void ServerApp::CheckForEmpireElimination() {
             entry.second->Eliminate();
         else {
             surviving_empires.insert(entry.second);
-            if (GetEmpireClientType(entry.second->EmpireID()) == Networking::CLIENT_TYPE_HUMAN_PLAYER)
+            if (GetEmpireClientType(entry.second->EmpireID()) != Networking::CLIENT_TYPE_AI_PLAYER)
```

Comment:
Could you add a comment explaining this? Presumably empires are controlled by AI clients, human empire players, or nobody (because the player isn't connected)? Are AI players always connected? Maybe change `surviving_human_empires` to `non_eliminated_non_ai_controlled_empires`?

Reply:
Added comment.
`ServerFSM::HandleNonLobbyDisconnection(const Disconnection& d)` stops the game if AI controls alive empire was disconnected.
Changed.